### PR TITLE
added cookie parser

### PR DESCRIPTION
**Pull request by Goldziher**
# PR Checklist

- [x] Have you followed the guidelines in `CONTRIBUTING.md`?
- [x] Have you got 100% test coverage on new code?
- [x] Have you updated the prose documentation?
- [x] Have you updated the reference documentation?

This PR adds a `cookie_string_parser` function. Its part of #612

### CODE REVIEW

**Comment by Goldziher** [comment]
related to #634

File: starlite/parsers.py [hunk 1]
```diff
@@ -78,3 +79,19 @@ def parse_form_data(media_type: "RequestEncodingType", form_data: "FormMultiDict
...
+    Args:
+        cookie_string: A cookie string.
+
+    Returns:
+        A string keyed dictionary of values
+    """
+    output: Dict[str, str] = {}
+    cookies = [cookie.split("=", 1) if "=" in cookie else ("", cookie) for cookie in cookie_string.split(";")]
+    for k, v in filter(lambda x: x[0] or x[1], ((k.strip(), v.strip()) for k, v in cookies)):
+        output[k] = unquote(unquote_cookie(v))
```
Comment (jtraub) [inline starlite/parsers.py:96]: Why do we use `urllib.parse.unquote` here on top of `http.cookies._unquote`? It seems Starlette simply returns values after applying `http.cookies._unquote`  as can be seen [here](https://github.com/encode/starlette/blob/master/starlette/requests.py#L54)

Comment (Goldziher) [reply starlite/parsers.py:96]: I saw, its copied verbatim from Django. But I then ran the code against a bunch of tests i created - based on tests for very popular cookie parsing tools in JS (40 million monthly downloads), and it failed for this value:

`("email=%20%22%2c%3b%2f", {"email": ' ",;/'})``

without using the urllib unquote there, the result of the value is: `{"email": ' "%20%22%2c%3b%2f'}`

**Review by jtraub — APPROVED** [review]

File: tests/test_parsers.py [hunk 2]
```diff
@@ -51,3 +54,25 @@ def test_parse_form_data() -> None:
...
+        ("  fizz; foo=  bar", {"": "fizz", "foo": "bar"}),
+        ("foo=false;bar=bar;foo=true", {"bar": "bar", "foo": "true"}),
+        ("foo=;bar=bar;foo=boo", {"bar": "bar", "foo": "boo"}),
+        (
+            Cookie(key="abc", value="123", path="/head", domain="localhost").to_header(header=""),
+            {"Domain": "localhost", "Path": "/head", "SameSite": "lax", "abc": "123"},
+        ),
+    ),
+)
+def test_cookie_parser(cookie_string: str, expected: Dict[str, str]) -> None:
```
Comment (provinzkraut) [inline tests/test_parsers.py:77]: Should probably be named `test_parse_cookie_string` if we're being pedantic :wink:

**Review by provinzkraut — APPROVED** [review]
Looking good!

File: tests/test_parsers.py (hunk 2 above)
Comment (Goldziher) [reply tests/test_parsers.py:77]: we are

**Review by provinzkraut — APPROVED** [review]

**Pull request merged**